### ISSUE TITLE
Multiple misspelled words in a file cause "on save" actions to time out

### ISSUE THREAD

**Issue by klanchman** [streetsidesoftware/vscode-spell-checker#349]
## Description

When VS Code is configured to run code actions when a file is saved, if a file has multiple spelling errors, code actions will time out and not be applied.

## Reproduction Steps

1. Set VS Code options:

```json
{
  "editor.codeActionsOnSave": {
    "source.fixAll": true
  },
  "editor.formatOnSave": true
}
```

4. Create a workspace with a TypeScript file in it, `test.ts`
5. Fill the file with the following contents:

```typescript
const test = () => {
  const foo = [
    'asdfasdfasdfasdfasdf',
    'asdfasdfasdfasdfasd',
    'asdfasdfasdfasdfas',
    'asdfasdfasdfasdfa',
    'asdfasdfasdfasdf',
    'asdfasdfasdfasd',
    'asdfasdfasdfas',
    'asdfasdfasdfa',
    'asdfasdfasdf',
    'asdfasdfasd',
  ]

  const formatMe   =    'formatting will not run'
}
```

4. Save the file

**Expected**: The file is formatted upon saving

**Actual**: The file is saved, but is not formatted

1. Remove all but 3 items in the array
2. Save the file - observe that formatting will run, but you may need to save the file more than 1 time to get it to happen

If I leave 4 items in the array, formatting is not applied. Three spelling errors seems to be the maximum where code actions will still sometimes be applied.

## Workarounds

I have found the following workarounds, but none are long term solutions

- Disable Code Spell Checker extension 😢
- Add `// spell-checker:disable` directive to file
  - Some other team members don't use a spell checker, let alone VS Code, so this would be noise for them
- Remove `"editor.codeActionsOnSave"` configuration
  - I have other code actions that I would like to run on save (e.g. TSLint)
- Whitelist all the misspelled words
  - The words _are_ actually misspelled, so I don't want to whitelist them, but I cannot change the words (data sent from external system)

## Other Notes

I use Prettier for formatting, but I was able to reproduce this issue using the built-in VS Code formatter as well.

## Versions

- macOS 10.14.5
- VS Code 1.34.0
- Code Spell Checker 1.7.12

**Comment by Jason3S** [comment]
@klanchman Excellent example!

**Comment by Jason3S** [comment]
@klanchman,
I took a look at this. Mainly because the spell checker does not listen for onSave events. I wanted to know how it would impact saving.

There are two different things going on here:
1. Applying all code actions (very expensive).
2. Formatting on save (also costly).

[VS Code API | onWillSaveTextDocument](https://code.visualstudio.com/api/references/vscode-api#workspace%2EonWillSaveTextDocument) says that there is a limited about of time given for the on save actions. If they are not completed in time, they will be ignored.

I have made a fix to ignore the code action request and to try and not spell check files while they are being saved.

**Comment by Jason3S** [comment]
You might need to increase the timeout:
```
	"editor.formatOnSaveTimeout": 5000,
```

**Comment by klanchman** [comment]
Hi @Jason3S, it looks like v1.7.15 is working as expected using my example file. I tried on my "actual" project as well, and it worked there too. Luckily I didn't need to change the `editor.formatOnSaveTimeout` setting, either. It might be my imagination, but it seems like formatting is being applied faster than before I had this issue!

Thanks for the quick response and fix! 😄
